src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -982,57 +952,8 @@ export class ConnectionDialogWebviewController extends FormWebviewController<
         });
     }
 
-    private async getConnectionGroups(): Promise<FormItemOptions[]> {
-        const rootId = this._mainController.connectionManager.connectionStore.rootGroupId;
-        let connectionGroups =
-            await this._mainController.connectionManager.connectionStore.readAllConnectionGroups();
-        connectionGroups = connectionGroups.filter((g) => g.id !== rootId);
-
-        // Count occurrences of group names to handle naming conflicts
-        const nameOccurrences = new Map<string, number>();
-        for (const group of connectionGroups) {
-            const count = nameOccurrences.get(group.name) || 0;
-            nameOccurrences.set(group.name, count + 1);
-        }
-
-        // Create a map of group IDs to their full paths
-        const groupById = new Map(connectionGroups.map((g) => [g.id, g]));
-
-        // Helper function to get parent path
-        const getParentPath = (group: IConnectionGroup): string => {
-            if (!group.parentId || group.parentId === rootId) {
-                return group.name;
-            }
-            const parent = groupById.get(group.parentId);
-            if (!parent) {
-                return group.name;
-            }
-            return `${getParentPath(parent)} > ${group.name}`;
-        };
-
-        const result = connectionGroups
-            .map((g) => {
-                // If there are naming conflicts, use the full path
-                const displayName = nameOccurrences.get(g.name) > 1 ? getParentPath(g) : g.name;
-
-                return {
-                    displayName,
-                    value: g.id,
-                };
-            })
-            .sort((a, b) => a.displayName.localeCompare(b.displayName));
-
-        return [
-            {
-                displayName: Loc.default,
-                value: rootId,
-            },
-            {
-                displayName: Loc.createConnectionGroup,
-                value: CREATE_NEW_GROUP_ID,
-            },
-            ...result,
-        ];
+    private async getConnectionGroups(mainController: MainController): Promise<FormItemOptions[]> {
+        return mainController.connectionManager.connectionStore.getConnectionGroupOptions();
```

Comment:
ConnectionStore is more for config operations, but this helper function is more about processing the names.  Maybe connectionManager or connectionUi would be a better place?